AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -247,44 +214,99 @@ - (void)readWithPartition:(NSString *)partition
+    // If not OK.
+    if (!data || [MSDataSourceError errorCodeFromError:cosmosDbError] !=
+        kMSACDocumentSucceededErrorCode) {
+      MSLogError([MSDataStore logTag], @"Not able to retrieve documents: %@",
+                 [cosmosDbError description]);
+      MSDataSourceError *dataSourceCosmosDbError =
+      [[MSDataSourceError alloc] initWithError:cosmosDbError];
+      MSPaginatedDocuments *documents =
+      [[MSPaginatedDocuments alloc] initWithError:dataSourceCosmosDbError];
+      completionHandler(documents);
+      return;
+    }
+    
+    // Deserialize.
+    NSError *deserializeError;
+    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:(NSData *)data
+                                                         options:0
+                                                           error:&deserializeError];
+    if (deserializeError) {
+      MSLogError([MSDataStore logTag], @"Error deserializing data: %@",
+                 [deserializeError description]);
+      MSDataSourceError *dataSourceDeserializeError =
+      [[MSDataSourceError alloc] initWithError:deserializeError];
+      MSPaginatedDocuments *documents =
+      [[MSPaginatedDocuments alloc] initWithError:dataSourceDeserializeError];
+      completionHandler(documents);
+      return;
+    }
+    MSLogDebug([MSDataStore logTag], @"Documents json: %@", json);
+    
+    // Get list of documents.
+    NSArray *jsonDocuments = json[@"Documents"];
+    NSMutableArray<MSDocumentWrapper *> *items = [NSMutableArray new];
+    for (id document in jsonDocuments) {
+      // TODO: handle deserialization errors here
+      
+      // Deserialize current document.
+      id<MSSerializableDocument> deserializedDocument =
+      [(id<MSSerializableDocument>)[documentType alloc]
+       initFromDictionary:(NSDictionary *)document];
+      
+      // Create a document wrapper object.
+      NSTimeInterval interval = [(NSString *)json[kMSDocumentTimestampKey] doubleValue];
+      NSDate *date = [NSDate dateWithTimeIntervalSince1970:interval];
+      NSString *eTag = json[kMSDocumentEtagKey];
+      MSDocumentWrapper *docWrapper =
+      [[MSDocumentWrapper alloc] initWithDeserializedValue:deserializedDocument
+                                                 partition:partition
+                                                documentId:@""
+                                                      eTag:eTag
+                                           lastUpdatedDate:date];
+      [items addObject:docWrapper];
+    }
+    
+    // Instanciate the first page and return it.
+    MSPage *page = [[MSPage alloc] initWithItems:items];
+    MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc]
+                                       initWithPage:page
+                                       partition:partition
+                                       documentType:documentType
+                                       readOptions:nil
```

Comment:
do you need to pass the readOptions?

Reply:
I think I am passing it down the line now. Cannot figure out how to look up which exact commit this comment was for :/